packages/adapters-library/src/SQLiteMetadataProvider.ts

```diff
@@ -43,109 +45,106 @@ export type PoolRow = {
   extra_reward_additional_data?: string // JSON stored as a string
 }
 
+const selectAllPoolsQuery = `
+  SELECT 
+    a.protocol_id,
+    a.product_id,
+    p.pool_id,
+    p.adapter_pool_id,
+    p.additional_data AS pool_additional_data,
+    t.token_address AS main_token_address,
+    t.token_name AS main_token_name,
+    t.token_symbol AS main_token_symbol,
+    t.token_decimals AS main_token_decimals,
+    ut.token_address AS underlying_token_address,
+    ut_t.token_name AS underlying_token_name,
+    ut_t.token_symbol AS underlying_token_symbol,
+    ut_t.token_decimals AS underlying_token_decimals,
+    ut.additional_data AS underlying_additional_data,
+    rt.token_address AS reward_token_address,
+    rt_t.token_name AS reward_token_name,
+    rt_t.token_symbol AS reward_token_symbol,
+    rt_t.token_decimals AS reward_token_decimals,
+    rt.additional_data AS reward_additional_data,
+    ert.token_address AS extra_reward_token_address,
+    ert_t.token_name AS extra_reward_token_name,
+    ert_t.token_symbol AS extra_reward_token_symbol,
+    ert_t.token_decimals AS extra_reward_token_decimals,
+    ert.additional_data AS extra_reward_additional_data
+  FROM  adapters a 
+        LEFT JOIN pools p ON a.adapter_id = p.adapter_id
+        LEFT JOIN tokens t ON p.pool_address = t.token_address
+        LEFT JOIN underlying_tokens ut ON p.pool_id = ut.pool_id
+        LEFT JOIN tokens ut_t ON ut.token_address = ut_t.token_address
+        LEFT JOIN reward_tokens rt ON p.pool_id = rt.pool_id
+        LEFT JOIN tokens rt_t ON rt.token_address = rt_t.token_address
+        LEFT JOIN extra_reward_tokens ert ON p.pool_id = ert.pool_id
+        LEFT JOIN tokens ert_t ON ert.token_address = ert_t.token_address;
```

Comment:
Query now returns everything (takes 60ms and it's only 16K rows, so no memory issues).

It then adds everything to in-memory objects that perform the same as the old json files once loaded.